lib/shopify/multipass.ex

```diff
@@ -0,0 +1,114 @@
+defmodule Shopify.Multipass do
+  @moduledoc """
+  SSO Logins can be accomplished by enabling the Multipass feature on your Shopify Plus account.
+  This requires the use of a special Multipass session token.
+  [https://help.shopify.com/en/api/reference/plus/multipass](https://help.shopify.com/en/api/reference/plus/multipass)
+  """
+
+  # The size of the crypto used by http://erlang.org/doc/man/crypto.html#block_encrypt-4
+  # Refer to the erlang docs as to which block sizes are allowed
+  @block_size 16
+
+  @doc ~S"""
+  Simple getter to expose the module attribute
+  """
+  def get_block_size(), do: @block_size
+
+  @doc ~S"""
+  Get the full URL for the Multipass signin.
+
+  ## Examples
+    iex> customer_data = %{
+    ...>     email: "something@test.shopify.com",
+    ...>     created_at: "2019-03-07T20:29:42.177390Z"
+    ...>   }
+    %{
+    created_at: "2019-03-07T20:29:42.177390Z",
+    email: "something@test.shopify.com"
+    }
+    iex> secret = "1234567890abcdef1234567890abcdef"
+    "1234567890abcdef1234567890abcdef"
+    iex>  url = Shopify.Multipass.get_url("myteststore", customer_data, secret)
+    "https://myteststore.myshopify.com/account/login/multipass/moaqEVx1Yu9hsvYvVpj-LeRYDtOo6ikicfTZd8tR8-xBMRg8tFjGEfllEcjj2VdbsezmT0XuEdglyQzi_biQPkfLJnP1dkxhNtfzwtt6IMQzu3W0qCPzbrUMD_gLaytPVP-zZZuYiSBqEMNdvzFg3zf0TOQHwbizX2D7It02sFI7ZpTRhfX4m_crV0b-DmmF"
+    iex>  url = Shopify.Multipass.get_url("", customer_data, secret, %{host: "custom-domain.com")
+    "https://custom-domain.com/account/login/multipass/moaqEVx1Yu9hsvYvVpj-LeRYDtOo6ikicfTZd8tR8-xBMRg8tFjGEfllEcjj2VdbsezmT0XuEdglyQzi_biQPkfLJnP1dkxhNtfzwtt6IMQzu3W0qCPzbrUMD_gLaytPVP-zZZuYiSBqEMNdvzFg3zf0TOQHwbizX2D7It02sFI7ZpTRhfX4m_crV0b-DmmF"
+  """
+  @spec get_url(binary, map, binary, map) :: binary
+  def get_url(shop_name, customer_data, multipass_secret, options \\ %{}) do
+    struct(
+      %URI{
+        host: "#{shop_name}.myshopify.com",
```

Comment:
When asking for the shop name, it is often unclear which version one should use: `myshop.myshopify.com` or just `myshop`. For that reason, there is a `scrub_shopify_name` function, that will return the shop name without `myshopify.com`. Since it's already there, we can use it here

```
# ...
host: Shopify.scrub_shop_name(shop_name) <> ".myshopify.com"
```

Reply:
Isn't that problematic for stores that use fully custom domain names? E.g. we have a production site that uses `somesite.com` as its Shopify domain.  So requests have to be made against that domain, and `.myshopify.com` must be omitted entirely.

Reply:
The shops I have worked with that use custom domains all still have their "original" shopify domain, and the admin page is not even redirected, so the API works w/o problems. This is my personal experience, so there is room for some error :bow:

I haven't really worked with any of the Gold Features, so this might actually be one of the cases, where you would want the custom URL, not the original one, to avoid an unnecessary redirect.
If that's the case, I'd suggest using the "raw input" for the host, and put an emphasis on having to use "my-shop.myshopify.com" or the custom domain in the Readme. 🤔